packages/eslint-plugin-wpcalypso/lib/rules/jsx-classname-namespace.js

```diff
@@ -1,23 +1,21 @@
 /**
- * @file Ensure JSX className adheres to CSS namespace guidelines
+ * @file Ensure JSX className adheres to BEM CSS naming conventions.
  * @author Automattic
- * @copyright 2016 Automattic. All rights reserved.
+ * @copyright 2022 Automattic. All rights reserved.
  * See LICENSE.md file in root directory for full license.
  */
 
 //------------------------------------------------------------------------------
 // Requirements
 //------------------------------------------------------------------------------
 
-const path = require( 'path' );
+const namePattern = new RegExp( `^[a-z0-9-]+(__[a-z0-9-]+)?$` );
```

Comment:
Is the BEM format we use documented somewhere? I see this regexp will match a class like `block__element` including a modified like `--foo`, if present. But it won't allow multiple `__` separators, but @jsnajdr's comment [here](p4TIVU-9Ok-p2#comment-11203) seems to suggest there might be multiple ones? I might be wrong, but thought I'd bring this up just in case!

Reply:
I didn't find any references to BEM convention specifically within wp-calypso, but MGS turned up some discussion of BEM. 

> @jsnajdr's comment [here](https://github.com/Automattic/wp-calypso/pull/p4TIVU-9Ok-p2#comment-11203) seems to suggest there might be multiple ones?

The link goes to 404. 
I decided on implementing the BEM convention as specified by the [working group](https://en.bem.info/methodology/naming-convention/) after reading the discussion around this rule in P2s.

Reply:
@fullofcaffeine I took a quick look to see how many files require the dual `__`, and looks like we only have 34 instances in wp-calypso:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/6549265/182962159-3be93c5c-0f14-4375-b24b-6ac62f392af9.png">

So I would say this particular exception/carve-out isn't too important for us to worry about.

Reply:
Will this mean we have to change all those?

Reply:
@noahtallen I think it would be good to have those classnames changed in the future to meet the updated guidelines introduced in this PR.

Reply:
Ah oops, I had merged the change already - was your comment intended to be blocking?

As far as I can tell, this change doesn't conflict with any other of our coding standards as I didn't observe new errors except for the ones cited above.

As for Gutenberg, I'm not as familiar with them and admit I haven't checked their coding standards before merging. I found [this page](https://developer.wordpress.org/block-editor/contributors/code/coding-guidelines/) which suggests Gutenberg also uses the BEM convention, and from what I can tell it does not appear that they permit the use of double __.

Reply:
👍 Sounds good. Nope, I didn't mean for it to be blocking! I just wanted to double check.